mlir/lib/Dialect/Tosa/Transforms/TosaValidation.cpp

```diff
@@ -147,107 +152,149 @@ struct TosaValidation : public tosa::impl::TosaValidationBase<TosaValidation> {
     return true;
   }
 
-  bool levelCheckRank(Operation *op, const Value &v,
-                      const std::string &checkDesc) {
+  bool levelCheckListSize(Operation *op, int32_t v,
+                          const std::string &checkDesc) {
+    if (v > tosaLevel.MAX_TENSOR_LIST_SIZE) {
+      op->emitOpError() << "failed level check for MAX_TENSOR_LIST_SIZE: "
+                        << checkDesc;
+      return false;
+    }
+    return true;
+  }
+
+  bool levelCheckRankAndSizes(Operation *op, const Value &v,
+                              const std::string &operandOrResult) {
     if (ShapedType type = dyn_cast<ShapedType>(v.getType())) {
       if (!type.hasRank()) {
         op->emitOpError() << "failed level check: unranked tensor";
```

Comment:
In some instances e.g. `CLAMP, ADD, TABLE, etc` rank 0 is supported, is this check too restrictive in this case?

Reply:
Rank 0 tensor also has rank but getRank() == 0. It aim to check unranked tensor that annotate as tensor<*xT>.

Reply:
Makes sense, thanks!